### PR TITLE
core: fix txs reindex after reindex from ancients crashed

### PR DESCRIPTION
**Pull request by ucwong**
once tx reindex crashing, it will not repair the txs index again

### CODE REVIEW

**Review by JurbinM — APPROVED** [review]

**Review by fjl — APPROVED** [review]

File: core/blockchain.go [hunk 1]
```diff
@@ -276,12 +276,12 @@ func NewBlockChain(db ethdb.Database, cacheConfig *CacheConfig, chainConfig *par
...
-		frozen, _ := bc.db.Ancients()
-		if frozen > 0 {
-			txIndexBlock = frozen
-		}
+	}
+	// If ancient database is not empty, reconstruct all missing
+	// indices in the background.
+	frozen, _ := bc.db.Ancients()
+	if frozen > 0 {
+		txIndexBlock = frozen
```
Comment (rjl493456442) [inline core/blockchain.go:284]: It's not correct, previously we will only re-init the ancient blocks in this special corner case where state is empty but ancient store it not. With this change we will re-init the ancient blocks all the time.

I will try to figure out a proper solution to fix it.

**Comment by rjl493456442** [comment]
Hi @ucwong , I have an alternative to rework the tx indexer https://github.com/ethereum/go-ethereum/pull/25723

I don't quite understand why the current version will not work after crash. Can you explain a bit more?

**Comment by ucwong** [comment]
> Hi @ucwong , I have an alternative to rework the tx indexer #25723
> 
> I don't quite understand why the current version will not work after crash. Can you explain a bit more?

The blockchain is not empty at all  , so ```txIndexBlock``` will alway be zero.

**Comment by holiman** [comment]
> It's not correct, previously we will only re-init the ancient blocks in this special corner case where state is empty but ancient store it not. With this change we will re-init the ancient blocks all the time.

plus superseded by #25723

**Pull request closed without merging**